extensions/amp-analytics/0.1/events.js

```diff
@@ -410,6 +525,149 @@ export class IniLoadTracker extends EventTracker {
 }
 
 
+/**
+ * Timer event handler.
+ */
+class TimerEventHandler {
+  /**
+   * @param {JsonObject} timerSpec The timer specification.
+   * @param {function(): UnlistenDef=} startBuilder Factory for building start
+   *     trackers for this timer.
+   * @param {function(): UnlistenDef=} stopBuilder Factory for building stop
+   *     trackers for this timer.
+   */
+  constructor(timerSpec, startBuilder, stopBuilder) {
+    /** @private {number|undefined} */
+    this.intervalId_ = undefined;
+
+    user().assert('interval' in timerSpec,
+        'Timer interval specification required');
+    /** @private @const {number} */
+    this.intervalLength_ = Number(timerSpec['interval']) || 0;
+    user().assert(this.intervalLength_ >= MIN_TIMER_INTERVAL_SECONDS,
+        'Bad timer interval specification');
+
+    /** @private @const {number} */
+    this.maxTimerLength_ = 'maxTimerLength' in timerSpec ?
+        Number(timerSpec['maxTimerLength']) : DEFAULT_MAX_TIMER_LENGTH_SECONDS;
+    user().assert(this.maxTimerLength_ > 0, 'Bad maxTimerLength specification');
+
+    /** @private @const {boolean} */
+    this.maxTimerInSpec_ = 'maxTimerLength' in timerSpec;
+
+    /** @private @const {boolean} */
+    this.callImmediate_ = 'immediate' in timerSpec ?
+        Boolean(timerSpec['immediate']) : true;
+
+    /** @private {?UnlistenDef} */
+    this.unlistenStart_ = null;
+
+    /** @private {?UnlistenDef} */
+    this.unlistenStop_ = null;
+
+    /** @private @const {function(): UnlistenDef|undefined} */
```

Comment:
Ideally, we get rid of `|undefined` part here and just deal with nullable types inside.

Reply:
Done.